tests/ui/pattern/slice-patterns-ambiguity.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(unused_variables)]
+
+struct Zeroes;
```

Comment:
not sure whether this test tests anything specific :thinking: might make snese to only have this as a check-pass test for irrefutable let?

please add the test separately where we incompletely a pattern as an array even though it should be a slice https://github.com/rust-lang/rust/issues/76342#issuecomment-1578787870

Reply:
> please add the test separately where we incompletely a pattern as an array even though it should be a slice https://github.com/rust-lang/rust/issues/76342#issuecomment-1578787870

Sorry, I don't understand what you're to say here. The quote does not seem to relate to the comment you link to.

Reply:
this test:
```rust
struct Foo(u32);
struct Bar;
impl Into<&'static Foo> for Bar {
    fn into(self) -> &'static Foo { &Foo(1) }
}
impl Into<Foo> for Bar {
    fn into(self) -> Foo { Foo(1) }
}

fn ty<T>(_: T) {
    println!("{}", std::any::type_name::<T>());
}

fn main() {
    let Foo(i) = Bar.into();
    ty(i); // u32
    let Foo(j) = <Bar as Into<&'static Foo>>::into(Bar);
    ty(j); // &u32
    let Foo(k): Foo = Bar.into();
    ty(k); // u32
    let Foo(l): &Foo = Bar.into();
    ty(l); // &u32
}
```

Reply:
woops, replacing `Foo` with an array :sweat:
```rust
#![allow(unused_variables)]
struct Zeroes;
impl Into<&'static [usize; 3]> for Zeroes {
    fn into(self) -> &'static [usize; 3] { &[0; 3] }
}
impl Into<[usize; 3]> for Zeroes {
    fn into(self) -> [usize; 3] { [0; 3] }
}
fn main() {
    let [a, b, c] = Zeroes.into();
    let [d, e, f] = <Zeroes as Into<&'static [usize; 3]>::into();
    let &[g, h, i] = Zeroes.into();
    let [j, k, l]: [usize; _] = Zeroes.into();
    let [m, n, o]: &[usize; _] = Zeroes.into();

}
```

Reply:
> `let [d, e, f] = <Zeroes as Into<&'static [usize; 3]>::into();`

Again not sure what you're trying to do here. This doesn't look like valid rust code to me. `into` takes a receiver.

Reply:
it can be valid by adding `Zeroes` as a receiver

```rust
    let [d, e, f] = <Zeroes as Into<&'static [usize; 3]>>::into(Zeroes);
```